sympy/solvers/bivariate.py

```diff
@@ -151,7 +154,13 @@ def _lambert(eq, x):
 
     # invert the generator X1 so we have x(u)
     u = Dummy('rhs')
-    xusolns = solve(X1 - u, x)
+    if domain is None:
+        # the flow from solve will proceed from here
+        domain_real = None
+        xusolns = solve(X1 - u, x)
+    else:
+        domain_real = domain.is_subset(S.Reals)
+        xusolns = solvify(X1 - u, x,domain)
```

Comment:
There should generally be a space after a comma (this applies elsewhere as well)